nightlies_watcher/treeherder.py

```diff
@@ -0,0 +1,42 @@
+import logging
+
+from thclient import TreeherderClient
+from nightlies_watcher.hg_mozilla import get_minimal_repository_name
+
+logger = logging.getLogger(__name__)
+
+client = TreeherderClient()
+
+
+def does_job_already_exist(repository, revision, job_name, tier=1):
+    minimal_repo_name = get_minimal_repository_name(repository)
+
+    resultsets = client.get_resultsets(
+        project=minimal_repo_name,
+        revision=revision,
+    )
+
+    if len(resultsets) == 0:
+        raise Exception('No result found for revision {} in repository {}'.format(revision, repository))
+    elif len(resultsets) != 1:
```

Comment:
Good questions! This part is about fetching a Treeherder result set (similar to push ids), like: https://treeherder.mozilla.org/api/project/mozilla-aurora/resultset/?count=10&full=true&revision=18f93e548a3cfaa67c4a852f81d14cb0714974b9

New builds will only add more jobs to the same result set. I don't know in which case 2 entries can come up, except in a buggy one.

I can take that check out, or comment in the code the reasoning